src/controllers/post.js

```diff
@@ -22,29 +22,39 @@ export const create = async (req, res) => {
 }
 
 export const getAll = async (req, res) => {
-  const getPosts = await getAllPostsDb()
-  const posts = getPosts.map((post) => {
-    const { user } = post
+  const commentsLimit = parseInt(req.query.comments) || undefined
 
-    if (!user.profile) {
-      return post
-    }
-    return {
+  try {
+    const getPosts = await getAllPostsDb()
```

Comment:
i'd rename getPosts to `postsData` for clarity

Reply:
No worries, i'll change it!